### PR TITLE
[addons] package size metadata

### PR DESCRIPTION
**Pull request by tamland**
Adds a `size` element (intended for repositories to set) which is the size of downloadable zip file. It can be displayed in a readable form with the `Listitem.AddonSize` into label.

This pr also includes some much needed db cleanup that should make it a bit less painful to add more metadata fields in the future.

### CODE REVIEW

File: xbmc/utils/StringUtils.cpp [hunk 1]
```diff
@@ -1284,3 +1284,29 @@ void StringUtils::Tokenize(const std::string& input, std::vector<std::string>& t
...
+  const std::array<std::string, 5> units{"B", "kB", "MB", "GB", "TB"};
+  if (bytes < 1000)
+    return Format("%luB", bytes);
+
+  int i = 0;
+  double value(bytes);
+  while (i < units.size() && value >= 999.5)
+  {
+    ++i;
+    value = bytes / std::pow(1024.0, i);
```
Comment (fritsch) [inline xbmc/utils/StringUtils.cpp]:
```
value /= 1024
```

would save the pow

**Comment by stefansaraev** [comment]
nice.

File: xbmc/utils/StringUtils.cpp (hunk 1 above)
Comment (stefansaraev) [inline xbmc/utils/StringUtils.cpp]: hmh.  999.5... ? and kB should be KB (those have different meanings)

File: xbmc/utils/StringUtils.cpp [hunk 2]
```diff
@@ -1284,3 +1284,29 @@ void StringUtils::Tokenize(const std::string& input, std::vector<std::string>& t
...
+  std::istringstream iss(str);
+  uint64_t result(fallback);
+  iss >> result;
+  return result;
+}
+
+std::string StringUtils::FormatFileSize(uint64_t bytes)
+{
+  const std::array<std::string, 5> units{"B", "kB", "MB", "GB", "TB"};
+  if (bytes < 1000)
```
Comment (stefansaraev) [inline xbmc/utils/StringUtils.cpp]: 1024

File: xbmc/utils/StringUtils.cpp (hunk 1 above)
Comment (tamland) [inline xbmc/utils/StringUtils.cpp]: @stefansaraev yes, 999.5. this is a smarter rounding that scales decimals with significance. See tests.

File: xbmc/utils/StringUtils.cpp (hunk 2 above)
Comment (tamland) [inline xbmc/utils/StringUtils.cpp]: same. intentional

File: xbmc/utils/StringUtils.cpp (hunk 1 above)
Comment (MartijnKaijser) [inline xbmc/utils/StringUtils.cpp]: maybe add line note to mention it's intentional?

**Comment by tamland** [comment]
Ok, added a docstring that should hopefully make it clearer how it's intended to work (there are also tests that shows the output of different sizes)

**Comment by tamland** [comment]
jenkins build this please

**Comment by tamland** [comment]
jenkins build this please

**Comment by HitcherUK** [comment]
@tamland Any chance of getting `ListItem.Property(Addon.Size)` added to the addon info dialog as well please?

**Comment by phil65** [comment]
I think it's listitem.addonsize and should already be available there?

**Comment by stefansaraev** [comment]
yes. $INFO[ListItem.AddonSize] worked for me when testing this PR.

**Comment by HitcherUK** [comment]
Looks like I've jumped the gun a bit as this isn't in the nightlies yet, thanks.

**Pull request merged**